src/pages/_app.tsx

```diff
@@ -1,26 +1,55 @@
 import {
-  Container,
   Experimental_CssVarsProvider as CssVarsProvider,
+  Stack,
 } from "@mui/material";
+import { Provider } from "react-redux";
 import { QueryClient, QueryClientProvider } from "react-query";
+import { getCookie } from "cookies-next";
+import { Container } from "@mui/system";
+import App from "next/app";
 
-import ThemeSwitcher from "../components/UI/ThemeSwitcher";
+import { api } from "../services/api";
+import { store } from "../store/index";
 import TopMenu from "../components/UI/TopMenu";
 
 const queryClient = new QueryClient();
 
 const MyApp = ({ Component, pageProps }) => {
+  api.defaults.headers.common["Authorization"] = `Bearer ${getCookie("token")}`;
+
   return (
-    <CssVarsProvider>
-      <QueryClientProvider client={queryClient}>
-        <ThemeSwitcher />
-        <Container maxWidth="md">
-          <TopMenu />
-          <Component {...pageProps} />
-        </Container>
-      </QueryClientProvider>
-    </CssVarsProvider>
+    <Provider store={store}>
+      <CssVarsProvider>
+        <QueryClientProvider client={queryClient}>
+          <TopMenu token={pageProps.token} />
```

Comment:
Let's move the jwt token to redux store instead of passing it as parameter everywhere